internal/controller/pipelinerollout_controller.go

```diff
@@ -196,9 +198,19 @@ func (r *PipelineRolloutReconciler) reconcile(
 				return false, err
 			}
 
-			kubernetes.SetAnnotation(pipelineRollout, apiv1.KeyHash, specHash)
+			kubernetes.SetAnnotation(pipelineRollout, apiv1.KeyHash, currentRolloutSpecHash)
 		}
 	} else {
+		// Calculate child resource spec hash only if no changes are necessary by the rollout spec comparison
+		pipelineSpecHash := ""
+		if currentRolloutSpecHash == pipelineRollout.Annotations[apiv1.KeyHash] {
+			pipelineSpecHash, err = calculateChildSpecHash(pipeline)
+			if err != nil {
+				numaLogger.Errorf(err, "failed to calculate spec hash from Pipeline CR: %v", err)
+				return false, err
+			}
+		}
+
```

Comment:
looks like all of the lifecycle management below is being done regardless of whether we need to be doing an update

Reply:
That is correct. We still want to perform certain operations independently from having to also perform an update (ex: pipeline pausing logic, ApplyCRSpec in case of creation, processPipelineStatus).

Reply:
Sorry, I specifically was referring to the pipeline pausing logic - that should only pertain if we are trying to perform an update of the spec, shouldn't it?

Reply:
Yes, in the next PRs (specifically when we will work on the updating feature) we should create functions to define when updates and pausing should be made based on the result of a function checking for specific constraints. In general, the logic below this line, and specifically that related to pausing, will be revisited.